### PR TITLE
Fix remove unused status of data config

### PR DESCRIPTION
**Pull request by dobleralex**
DdiConfigData falsely extended DdiActionFeedback which forced it to add an `"status"` property to every request although its value was never used within code. This PR removes the unnecessary fields of DdiActionFeedback. As it is set to ignore unknown properties, old requests that contain the `"status"` property will still work, maintaining backwards compatibility.

All test were also adapted accordingly to not use the `"status"` property, it was also removed from the documentation tests. In addition a test was added to ensure backwards compatibility is not broken by checking if a request with the `"status"` property still works as before.

### CODE REVIEW

File: hawkbit-rest/hawkbit-ddi-api/src/test/java/org/eclipse/hawkbit/ddi/json/model/DdiConfigDataTest.java [hunk 1]
```diff
@@ -37,56 +37,52 @@
...
+    public void shouldDeserializeObjectWithStatusProperty() throws IOException {
+        // Setup
+        String serializedDdiConfigData = "{\"id\":123,\"time\":\"20190809T121314\","
+                + "\"status\":{\"execution\":\"closed\",\"result\":{\"finished\":\"success\",\"progress\":null},"
+                + "\"details\":[]},\"data\":{\"test\":\"data\"},\"mode\":\"replace\"}";
+
+        // Test
+        DdiConfigData ddiConfigData = mapper.readValue(serializedDdiConfigData, DdiConfigData.class);
+
+        assertThat(ddiConfigData.getMode()).isEqualTo(DdiUpdateMode.REPLACE);    }
```
Comment (Nkyn) [inline hawkbit-rest/hawkbit-ddi-api/src/test/java/org/eclipse/hawkbit/ddi/json/model/DdiConfigDataTest.java:87]: - Move closing bracket } to new line.
- Add newline to end of file.

Comment (dobleralex) [reply hawkbit-rest/hawkbit-ddi-api/src/test/java/org/eclipse/hawkbit/ddi/json/model/DdiConfigDataTest.java]: fixed

File: hawkbit-rest/hawkbit-ddi-api/src/test/java/org/eclipse/hawkbit/ddi/json/model/DdiConfigDataTest.java [hunk 2]
```diff
@@ -37,56 +37,52 @@
...
-                + "\"details\":[]},\"data\":{\"test\":\"data\"},\"mode\":\"replace\",\"unknownProperty\":\"test\"}";
+        String serializedDdiConfigData = "{\"data\":{\"test\":\"data\"},\"mode\":[\"replace\"],\"unknownProperty\":\"test\"}";
 
         // Test
         mapper.readValue(serializedDdiConfigData, DdiConfigData.class);
     }
+
+    @Test
+    @Description("Verify the correct deserialization of a model with removed unused status property")
+    public void shouldDeserializeObjectWithStatusProperty() throws IOException {
```
Comment (Nkyn) [inline hawkbit-rest/hawkbit-ddi-api/src/test/java/org/eclipse/hawkbit/ddi/json/model/DdiConfigDataTest.java:77]: Add reason/description for this test:
properties were there before. But are gone now.

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Pull request merged**